examples/99-advanced/atomic-orbitals.py

```diff
@@ -0,0 +1,226 @@
+#
+#    .. code:: python
+#
+#       pip install sympy
+
+import numpy as np
+
+import pyvista as pv
+from pyvista import examples
+
+###############################################################################
+# Generate the Dataset
+# ~~~~~~~~~~~~~~~~~~~~
+# Generate the dataset by evaluating the analytic hydrogen wave function from
+# ``sympy``.
+#
+# .. math::
+#    \begin{equation}
+#        \label{eq:hydro_wfc}
+#        \psi_{nlm}(r,\theta,\phi)
+#        =
+#        \sqrt{
+#            \left(\frac{2}{na_0}\right)^3\, \frac{(n-l-1)!}{2n[(n+l)!]}
+#        }
+#        e^{-r / na_0}
+#        \left(\frac{2r}{na_0}\right)^l
+#        L_{n-l-1}^{2l+1} \cdot Y_l^m(\theta, \phi)
+#    \end{equation}
+#
+# See `Hydrogen atom <https://en.wikipedia.org/wiki/Hydrogen_atom>`_ for more
+# details.
+#
+# This dataset evaluates this function for the hydrogen orbital 3dxy, with the
+# following quantum numbers:
+#
+# * Principal quantum number: ``n=3``
+# * Azimuthal quantum number: ``l=2``
+# * Magnetic quantum number: ``m=-2``
+
+grid = examples.load_hydrogen_orbital(3, 2, -2)
+grid
+
+
+###############################################################################
+# Plot the Orbital
+# ~~~~~~~~~~~~~~~~
+# Plot the orbital using :func:`add_volume() <pyvista.Plotter.add_volume>` and
+# using the default scalars contained in ``grid``, ``real_hwf``. This way we
+# can plot more than just the probability of the election, but also the phase
+# of the electron.
+#
+# .. note::
+#    Since the real value of evaluated wave function for this orbital varies
+#    between ``[-<value>, <value>]``, we cannot use the default opacity
+#    ``opacity='linear'``. Instead, we use ``[1, 0, 1]`` since we would like
+#    the opacity to be proportional to the absolute value of the scalars.
+
+pl = pv.Plotter()
+vol = pl.add_volume(grid, cmap='magma', opacity=[1, 0, 1])
+vol.prop.interpolation_type = 'linear'
+pl.camera.zoom(1.5)
+pl.show_axes()
+pl.show()
+
+
+###############################################################################
+# Plot the Orbital Contours as an Isosurface
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Generate the contour plot for the orbital by determining when the orbital
+# equals 10% the maximum value of the orbital. This effectively captures the
+# majority of the "volume" the electron potentially exists in.
+#
+# Note how we use the absolute value of the scalars when evaluating
+# :func:`contour() <pyvista.PolyDataFilters.contour>`. Otherwise, we would only
+# capture the location of the spin-up electron.
+
+eval_at = grid['real_hwf'].max() * 0.1
+contours = grid.contour(
+    [eval_at],
+    scalars=np.abs(grid['real_hwf']),
+    method='marching_cubes',
+)
+contours = contours.interpolate(grid)
+contours.plot(
+    categories=2,
+    smooth_shading=True,
+    annotations={-eval_at: 'down-spin', eval_at: 'up-spin'},
+    scalar_bar_args={'n_labels': 0, 'title': ''},
+)
```

Comment:
Now this is a conceptual bug: there's no spin here. On each of these orbitals (so e.g. on the 3d_{xy} orbital) you could place 2 electrons without violating Pauli's exclusion principle: an up-spin state and a down-spin state are different quantum state so they can coexist. If you've seen orbital energy diagrams in chemistry class (I know I have, but certainly many people don't):

![electron configuration of Br](https://periodictable.me/wp-content/uploads/2018/09/2000px-Electron_configuration_bromine.svg_.png)

This is the electronic structure of atomic bromine. In the top right you can see 5 horizontal lines above "3d", these are the 5 3d orbitals (ranging from m = -2 to 2). On each of these orbitals you have an up and a down arrow: two electrons!

So my point is that a single orbital (for a single electron) occupies all four lobes of the wave function. What is plotted is the phase of the wave function, you can think of it as a sign (as in +-1). In some places it's negative, in some places it's positive. But this information is lost when you take the square of the wave function in order to obtain a probability density (which is non-negative).

Reply:
Thanks for the explanation!

Reply:
My pleasure, but as I said this is a "bug" in the example which we should fix, spin should not be mentioned at all. Unresolving this.

Reply:
Ah, sorry, github fooled me by only showing some of the commits. Spin seems gone now, thanks.